### PR TITLE
Add root hash consistency check

### PR DESCRIPTION
**Pull request by hojoung97**
I added a very basic root hash consistency check.
1. Grabs the latest log consistency proof from Rekor server
2. Checks if the last seen root hash is part of the latest log consistency proof from step 1

For now, I have simply queried the latest log information to check it against the latest log consistency proof.
Moving forward we can probably temporarily use a text file with latest root hash saved until we actually run the monitor's transparency log that will hold the latest root hash audited.

### CODE REVIEW

File: mirroring/mirroring.go [hunk 1]
```diff
@@ -90,6 +91,42 @@ func GetLogInfo() (*models.LogInfo, error) {
 	return logInfoResp.GetPayload(), nil
 }
 
+func GetLogProof(firstSize *int64) (*models.ConsistencyProof, error) {
+	rekorServerURL := viper.GetString("rekorServerURL")
+	rekorClient, err := client.GetRekorClient(rekorServerURL)
```
Comment (SantiagoTorres) [inline mirroring/mirroring.go:96]: Hmm, I wonder if we'd like to split out the CLI/config setting parse routines from this function and make the server url a parameter instead...

File: mirroring/mirroring.go [hunk 2]
```diff
@@ -90,6 +91,42 @@ func GetLogInfo() (*models.LogInfo, error) {
...
+	params.LastSize = *logInfo.TreeSize
+
+	logProofResp, err := rekorClient.Tlog.GetLogProof(params)
+	if err != nil {
+		return nil, err
+	}
+	return logProofResp.GetPayload(), nil
+}
+
+func VerifyConsistencyProof(rootHash *string, logProof *models.ConsistencyProof) bool {
```
Comment (SantiagoTorres) [inline mirroring/mirroring.go:119]: I'm not sure if this function is doing all it should, but we can discuss this. Shouldn't it be reconstructing the new roothash using the old hash and the proof

**Review by SantiagoTorres — CHANGES_REQUESTED** [review]
I think there's some bits that I'd like to discuss, but this is looking in the right direction!

File: mirroring/mirroring.go (hunk 1 above)
Comment (hojoung97) [reply mirroring/mirroring.go:96]: Yes, I had somewhat related doubt which is the fact that we are initializing a new rekor client for each operation we perform. We can make the server URL as a parameter but what do you think of making a rekor client instance as a parameter?

Comment (SantiagoTorres) [reply mirroring/mirroring.go:96]: Hmm, good point. Let's do the latter :+1:

**Comment by hojoung97** [comment]
I made some changes here.
I want to just highlight that the root hash consistency check is now computing a new root hash from last seen root hash and compares computed hash against the newly advertised root hash. 
For now, I inserted a very basic testing with the last tree size being 1.

**Review by SantiagoTorres — APPROVED** [review]
Fantastic work!

**Comment by SantiagoTorres** [comment]
FWIW I liked your use of the rfc6962 module. IT keeps things clear and succint :)

**Pull request merged**